libscript/src/objc.lcb

```diff
@@ -610,4 +610,212 @@ public handler CreateObjcInformalDelegateWithContext(in pProtocol as List, in pH
     end unsafe
 end handler    
 
+/* Block handling */
+
+private variable mNSConcreteGlobalBlock as optional Pointer
+private constant kRTLD_LAZY is 1
+private type LibraryHandle is Pointer
+
+private foreign handler OpenLibrary(in pPath as ZStringUTF8, in pFlag as CSInt) \
+	returns optional LibraryHandle \
+	binds to "dlopen"
+
+private foreign handler GetSymbolPointer(in pHandle as LibraryHandle, in pSymbol as ZStringUTF8) \
+	returns optional Pointer \
+	binds to "dlsym"
+
+private foreign handler CloseLibrary(in pHandle as LibraryHandle) \
+	returns CSInt \
+	binds to "dlclose"
+
+/* Fetch the `isa` pointer for the (internal) Obj-C concreate block class */
+private unsafe handler EnsureNSConcreteGlobalBlock() returns nothing
+	if mNSConcreteGlobalBlock is not nothing then
+		return
+	end if
+
+	/* As we cannot bind to the pointer to a symbol as a variable, we use
+	 * dynamic library loading and resolution functions to do so directly. */
+	variable tSystem as LibraryHandle
+	put OpenLibrary("libSystem.dylib", kRTLD_LAZY) into tSystem
+	if tSystem is not nothing then
+		put GetSymbolPointer(tSystem, "_NSConcreteGlobalBlock") \
+			into mNSConcreteGlobalBlock
+		CloseLibrary(tSystem)
+	end if
+end handler
+
+private constant kBlockSize is 32
+private constant kBlockDescriptorSize is 40
+
+public foreign type _ObjcBlockDescriptor binds to "MCAggregateTypeInfo:ffrrr"
+public foreign type _ObjcBlock binds to "MCAggregateTypeInfo:rEErr"
+public type ObjcBlockPointer is Pointer
+
+foreign handler MCHandlerGetFunctionPtr(in pHandler as any, out rPtr as Pointer) \
+	returns CBool \
+	binds to "<builtin>"
+
+private foreign handler CopyBlockDescriptor(inout pBlock as _ObjcBlockDescriptor, in pSize as UIntSize, out rNewBlock as Pointer) \
+	returns CBool \
+	binds to "MCMemoryAllocateCopy"
+
+private foreign handler CopyBlock(inout pBlock as _ObjcBlock, in pSize as UIntSize, out rNewBlock as Pointer) \
+	returns CBool \
+	binds to "MCMemoryAllocateCopy"
+
+private foreign handler MCMemoryDelete(in pBlock as Pointer) \
+	returns nothing \
+	binds to "<builtin>"
+
+private foreign handler BlockPointerToBlock(out rBlock as _ObjcBlock, in pPointer as ObjcBlockPointer, in pSize as UIntSize) \
+	returns nothing \
+	binds to "c:memcpy"
+
+/**
+Create an Objective-C block pointer that wraps an LCB handler.
+
+Example:
```

Comment:
I take it 'example' blocks in the docs comments are automagically made code-like in whatever output they are generated to?

Reply:
Having now got the dictionary to build correctly with this patch, I can confirm they are automatically formatted as code blocks : )